### ISSUE TITLE
Extremely slow network within containers, pulling images fails, packet loss

### ISSUE THREAD

**Issue by siosphere** [docker/for-mac#1820]
### Expected behavior
The network should be as fast as the host, should be able to pull images without a timeout
### Actual behavior
The network is extremely slow, pinging takes several seconds, and seems to randomly go up and down on response time
Trying to pull images results in a client timeout while awaiting headers

### Information
Diagnostic ID: 32AAB3E8-2185-476F-83A4-C972706FAE4A
```
Docker for Mac: version: 17.06.0-ce-mac18 (4cdec4294a50b2233146b09469b49937dabdebdd)
macOS: version 10.11.6 (build: 15G31)
logs: /tmp/32AAB3E8-2185-476F-83A4-C972706FAE4A/20170703-151904.tar.gz
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     db
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     menubar
[OK]     disk
```

Docker info:
```Containers: 38
 Running: 17
 Paused: 0
 Stopped: 21
Images: 176
Server Version: 17.06.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 515
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.31-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 6
Total Memory: 11.71GiB
Name: moby
ID: B65G:6JQ7:DDOM:4LCM:52FY:GZ3J:5AFR:SUDK:TGTG:R2AH:7CXH:2WV2
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 153
 Goroutines: 131
 System Time: 2017-07-03T22:23:51.40853915Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

### Steps to reproduce the behavior
Restarted OSX
Attempted to pull image, timed out
```
docker pull alpine:3.6
Error response from daemon: Get https://registry-1.docker.io/v2/library/alpine/manifests/3.6: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Falpine%3Apull&service=registry.docker.io: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```
Inside of container attempt to ping, and get:
```
docker run --rm -it centos:7 ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=37 time=1941 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=37 time=883 ms
64 bytes from 8.8.8.8: icmp_seq=3 ttl=37 time=4959 ms
64 bytes from 8.8.8.8: icmp_seq=4 ttl=37 time=3879 ms
64 bytes from 8.8.8.8: icmp_seq=5 ttl=37 time=2837 ms
64 bytes from 8.8.8.8: icmp_seq=6 ttl=37 time=1799 ms
64 bytes from 8.8.8.8: icmp_seq=7 ttl=37 time=757 ms
64 bytes from 8.8.8.8: icmp_seq=8 ttl=37 time=5215 ms
64 bytes from 8.8.8.8: icmp_seq=9 ttl=37 time=4139 ms
64 bytes from 8.8.8.8: icmp_seq=10 ttl=37 time=3098 ms
64 bytes from 8.8.8.8: icmp_seq=11 ttl=37 time=2059 ms
64 bytes from 8.8.8.8: icmp_seq=12 ttl=37 time=1020 ms
--- 8.8.8.8 ping statistics ---
13 packets transmitted, 12 received, 7% packet loss, time 12477ms
rtt min/avg/max/mdev = 757.373/2715.975/5215.803/1493.468 ms, pipe 5
```
pinging from host is unaffected:
```
ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8): 56 data bytes
64 bytes from 8.8.8.8: icmp_seq=0 ttl=60 time=6.601 ms
64 bytes from 8.8.8.8: icmp_seq=1 ttl=60 time=14.829 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=60 time=8.141 ms
64 bytes from 8.8.8.8: icmp_seq=3 ttl=60 time=6.852 ms
64 bytes from 8.8.8.8: icmp_seq=4 ttl=60 time=7.217 ms
64 bytes from 8.8.8.8: icmp_seq=5 ttl=60 time=5.621 ms
64 bytes from 8.8.8.8: icmp_seq=6 ttl=60 time=5.684 ms
--- 8.8.8.8 ping statistics ---
7 packets transmitted, 7 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 5.621/7.849/14.829/2.962 ms
```

**Comment by siosphere** [comment]
This happens when there is a container running that has a socket connection.

This did not happen prior to 17.06.0-ce-mac18. Even with absolutely no network traffic going through the socket, any new container, existing container, etc, will experience the above slowdown.

**Comment by joaquincasares** [comment]
I've also been experiencing similar issues, even before the recent update. I usually use `docker-compose build`, but running simple `apt-get update` and `apt-get install -y git` on the `openjdk:8-jdk` slowed to a crawl.

I tried `Remove all data` and `Reset to factory defaults` but the issue was unresolved. After uninstalling Docker for Mac, downloading a new .dmg, and reinstalling the problem seemed to resolve itself. So the issue may lie within the steps that are included in the uninstall workflow and are not included in the factory reset workflow.

Next time I'll run the `Diagnose and Feedback` process now that I learned what that button does. :)

**Comment by SGrondin** [comment]
I've been experiencing similar issues, long-lived sockets (such as ZMQ) seemed to be more affected than short-lived ones like HTTP. 5-10% of all calls would timeout.

Running the [Docker For Mac Diagnostics](https://docs.docker.com/docker-for-mac/troubleshoot/) didn't find any issue, but fixed the problem. Weird.

**Comment by chdeliens** [comment]
> I've been experiencing similar issues, long-lived sockets (such as ZMQ) seemed to be more affected than short-lived ones like HTTP. 5-10% of all calls would timeout.
> 
> Running the [Docker For Mac Diagnostics](https://docs.docker.com/docker-for-mac/troubleshoot/) didn't find any issue.. but it fixed the problem! Weird.

Same here, just running the diagnostics fixed the issue! At least for now... 😆

**Comment by fourgates** [comment]
I have experienced the same issue.

Diagnosis ID: 54A0E2A5-DFEA-4380-8AAC-AABF6C10E66B/20200124232052

I use Docker on my work PC with no issues. I recently started developing on an iMac at home and it takes significantly longer to serve up the same main.js file. No issues when working on the PC. Same code, same images/containers, and I think my iMac actually has a much more powerful processor and more RAM.
